errors.go

```diff
@@ -0,0 +1,63 @@
+package influxdb
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// ErrUnimplemented is an error for when pieces of the client's functionality is unimplemented.
+var ErrUnimplemented = errors.New("unimplemented")
+
+type maxRetriesExceededError struct {
+	tries int
+	err   error
+}
+
+func (err maxRetriesExceededError) Error() string {
+	return fmt.Sprintf("max retries of %d reached, and we recieved an error of %v", err.tries, err.err)
+}
+
+func (err maxRetriesExceededError) Unwrap() error {
+	return err.err
+}
+
+type coalescingError []error
+
+func (err coalescingError) Error() string {
+	buf := strings.Builder{}
+	for _, e := range []error(err) {
+		if e == nil {
+			continue
+		}
+		buf.WriteString(e.Error())
+	}
+	return buf.String()
+}
+
+func (err coalescingError) Unwrap() error {
+	if len(err) > 1 {
+		return coalescingError(err[1:])
+	}
+	if len(err) == 1 {
+		return err[0]
+	}
+	return nil
+}
+
+type genericRespError struct {
+	Code      string
```

Comment:
Are we sure this is `Code` or the `Status`? In the usage in your client you assign the `resp.Status` to this field instead of the `resp.StatusCode`. I would make them consistent for clarity.

Reply:
Yes it is Code and a string.

The db returns an error in the form of a json string.
Example:
```json
{"code":"unauthorized","message":"unauthorized access"}
```